### PR TITLE
Fixes after expo

### PR DESCRIPTION
**Pull request by jonzee**
- small fixes after merging pr's for expo
- updated dependencies (react, formsy, material-ui)
- created `multiselect` with group toggle (and tests)

### CODE REVIEW

File: src/ts/Detail/Offerer.tsx [hunk 1]
```diff
@@ -26,10 +26,10 @@ export default function Offerer({ offer }) {
           <div className="offerer-main">
             <div>{name}</div>
             <div>{`${city}, ${country}`}</div>
-            <div>{`${i('With VG since')} ${moment(offerer.since).format('DD MMM YYYY')}`}</div>
+            <div>{`${i('With VG since')} ${moment(since).format('DD MMM YYYY')}`}</div>
```
Comment (michalmikolajczyk) [inline src/ts/Detail/Offerer.tsx]: what do you think of assigning those calls to constants?

File: src/ts/Detail/ListVG.tsx [hunk 2]
```diff
@@ -8,40 +8,40 @@ import parser from '../helpers/parser';
...
+    schema: 'services',
+    where: 'services',
   },
 ];
 
 export default function ListVG({ offer }) {
   const accordions = _.map(tabs, (tab, index) => {
     const items = _.get(offer, tab.where);
-    if (items === null || items[0] === undefined) return null;
+    if (items === undefined || items[0] === undefined) return null;
```
Comment (michalmikolajczyk) [inline src/ts/Detail/ListVG.tsx]:
```
if (!items || !items[0]) return null
```

File: src/ts/Search/Select.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+    expect([...instance.state.values]).to.not.contain('Motorcycle');
+    expect([...instance.state.values]).to.be.deep.equal(items.slice(1).slice(0, -1));
+    instance.setState = sinon.spy(instance.setState);
+    expect(instance.setState).to.not.have.been.called;
+    instance['onChange'](undefined, undefined, items.slice(1));
+    expect(instance.setState).to.have.been.calledOnce;
+    expect([...instance.state.values]).to.contain('Motorcycle');
+  });
+});
```
Comment (michalmikolajczyk) [inline src/ts/Search/Select.spec.tsx]: 👍

File: src/ts/Search/Select.tsx [hunk 4]
```diff
@@ -1,134 +1,166 @@
...
 import i from '../i18n';
 
 export default class Select extends React.Component<any, any> {
-  private dataSource = [];
-  constructor(props) {
-    super(props);
-    // select keeps values chosen by user
-    this.state = { value: [] };
-    this.onChange = this.onChange.bind(this);
+  state = {
```
Comment (michalmikolajczyk) [inline src/ts/Search/Select.tsx]: ??

File: src/ts/Search/Select.tsx [hunk 5]
```diff
@@ -1,134 +1,166 @@
...
+  };
 
-    const createGroup = (values, group) => (
-      values.map((value, index) => (
-        <div
-          key={index}
-          label={value}
-          value={`${group}#${value}`}
-          className="select-item"
+  selectionRenderer = (values) => {
```
Comment (michalmikolajczyk) [inline src/ts/Search/Select.tsx]: why would we define a method that way?

File: src/ts/helpers/test-theme.tsx [hunk 6]
```diff
@@ -14,6 +14,17 @@ export function mountWithTheme(node) {
   });
 }
 
+export function muiContext() {
```
Comment (michalmikolajczyk) [inline src/ts/helpers/test-theme.tsx]: 👍

File: src/ts/Detail/Offerer.tsx (hunk 1 above)
Comment (jonzee) [inline src/ts/Detail/Offerer.tsx]: good point

File: src/ts/Search/Select.tsx (hunk 4 above)
Comment (jonzee) [inline src/ts/Search/Select.tsx]: I don't need `constructor` to perform any other operation, so I define `state` as prop, it's compile to the same code: http://stackoverflow.com/questions/37788342

File: src/ts/Search/Select.tsx (hunk 5 above)
Comment (jonzee) [inline src/ts/Search/Select.tsx]: when I use arrow function I don't need to `bind` methods in `constructor`, it looks like the best solution https://medium.com/@housecor/react-binding-patterns-5-approaches-for-handling-this-92c651b5af56

File: src/ts/Search/Select.tsx (hunk 4 above)
Comment (michalmikolajczyk) [inline src/ts/Search/Select.tsx]: setting the state in the constructor is the recommended way: https://facebook.github.io/react-native/docs/state.html

I understand that they MAY be equivalent (?), but let's just keep a single way of handling things please

File: src/ts/Search/Select.tsx (hunk 5 above)
Comment (michalmikolajczyk) [inline src/ts/Search/Select.tsx]: do we have Babel stage 2 preset on, to make this work? http://stackoverflow.com/questions/36677733/why-jsx-props-should-not-use-arrow-functions

Comment (jonzee) [inline src/ts/Search/Select.tsx]: we use Babel preset `stage-0`, so it's includes `stage-2` https://babeljs.io/docs/plugins/preset-stage-0/

**Pull request merged**